components/tidb_query_executors/src/runner.rs

```diff
@@ -606,7 +624,12 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
         }
 
         warnings.merge(&mut result.warnings);
-        Ok((is_drained, record_len))
+        let wait = self.quota_limiter.consume_read(
+            1,
+            record_len,
+            (thread_cpu_time.elapsed().as_micros()) as usize,
```

Comment:
It is costy to fetch the thread cpu time. Now, we read thread cpu time no matter whether this feature is on or off. I believe this will cause performance regression, especially for point get.

Reply:
In `QuotaLimiter`, if it set to 0, limiter will jump and just return Duration::ZERO

Reply:
But most of the cost comes from `ThreadTime::now()` and `thread_cpu_time.elapsed()`, which are not filtered by the feature flag.

Reply:
Got it